### PR TITLE
Nvyn

### PR DESCRIPTION
**Pull request by NazarVynnyk**
Add funcktion for deleting products from store, changed validation

### CODE REVIEW

**Comment by NazarVynnyk** [comment]
Add funcktion for deleting products from store, changed validation

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/ExceptionHandlerController.java [hunk 1]
```diff
@@ -31,7 +33,7 @@
 
     @ExceptionHandler(HttpClientErrorException.class)
     public String handleRestClientException(HttpClientErrorException e, final RedirectAttributes redirectAttributes) {
-        LOGGER.error(e.getMessage());
+        LOGGER.error(e.getMessage(), Arrays.toString(e.getStackTrace()));
```
Comment (kucherif) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/ExceptionHandlerController.java]: `LOGGER.error(e.getMessage(), e);`
Doesn't work?

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/StorePageController.java [hunk 2]
```diff
@@ -202,9 +206,34 @@ public String addProductsToStore(@RequestParam("storeId") String storeId, @Model
...
+            productID) {
+
+        final String productUri = storeUrl + "/{storeId}/products/{productId}";
+        final String delUri = storeUrl + "/" + storeID + "/products";
+
+        RestTemplate restTemplate = getRestTemplate();
+        Map<String, Integer> param = new HashMap<>();
+        param.put("storeId", storeID);
+        param.put("productId", productID);
+        Product product = restTemplate.getForObject(productUri, Product.class, param);
```
Comment (kucherif) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/StorePageController.java]: Don't forget to move this to the service

File: MVCapplication/src/main/webapp/WEB-INF/resources/js/storeTable.js [hunk 3]
```diff
@@ -24,6 +27,29 @@ $("#dialog").dialog({autoOpen:false,buttons:{
...
+    Delete:function(){
+        $.ajax({
+            url : "/stores/delProduct",
+            method : "POST",
+            data : {
+                storeID: storId,
+                 productID: prodId}
+        });
+        $(this).dialog("close");
+        setTimeout(function() {window.location.reload();}, 1000);
```
Comment (kucherif) [inline MVCapplication/src/main/webapp/WEB-INF/resources/js/storeTable.js]: you do not need ajax here. use plain request from jsp

File: RESTserver/src/main/java/com/softserve/if072/restservice/controller/ExceptionHandlerController.java [hunk 4]
```diff
@@ -22,30 +24,30 @@
     @ExceptionHandler(DataNotFoundException.class)
     @ResponseStatus(HttpStatus.NOT_FOUND)
     public String dataNotFound(DataNotFoundException e) {
-        LOGGER.error(e.getMessage());
+        LOGGER.error(e.getMessage(), Arrays.toString(e.getStackTrace()));
```
Comment (kucherif) [inline RESTserver/src/main/java/com/softserve/if072/restservice/controller/ExceptionHandlerController.java]: the same as above

**Pull request merged**